Core/src/org/sleuthkit/autopsy/geolocation/datamodel/Area.java

```diff
@@ -0,0 +1,177 @@
+ */
+public final class Area extends GeoPath {
+    private static final Logger LOGGER = Logger.getLogger(Track.class.getName());
+
+    /**
+     * Construct a new Area for the given artifact.
+     *
+     * @param artifact
+     *
+     * @throws GeoLocationDataException
+     */
+    public Area(BlackboardArtifact artifact) throws GeoLocationDataException {
+        this(artifact, Waypoint.getAttributesFromArtifactAsMap(artifact));
+    }
+
+    /**
+     * Construct an Area for the given artifact and attributeMap.
+     *
+     * @param artifact     TSK_GPD_TRACK artifact
+     * @param attributeMap Map of the artifact attributes
+     *
+     * @throws GeoLocationDataException
+     */
+    private Area(BlackboardArtifact artifact, Map<BlackboardAttribute.ATTRIBUTE_TYPE, BlackboardAttribute> attributeMap) throws GeoLocationDataException {
+        super(artifact, getAreaName(attributeMap));
+
+        GeoAreaPoints points = getPointsList(attributeMap);
+        buildPath(points, artifact);
+    }
+
+    /**
+     * Return the name of the area from the attributeMap. Track name is stored
+     * in the attribute TSK_NAME
+     *
+     * @param attributeMap
+     *
+     * @return Area name or empty string if none was available.
+     */
+    private static String getAreaName(Map<BlackboardAttribute.ATTRIBUTE_TYPE, BlackboardAttribute> attributeMap) {
+        BlackboardAttribute attribute = attributeMap.get(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_NAME);
+
+        return attribute != null ? attribute.getValueString() : "";
+    }
+
+    /**
+     * Create the list of AreaWaypoints from the GeoTrackPoint list.
+     *
+     * @param points   GeoAreaPoints object.
+     * @param artifact The artifact to which these points belong
+     *
+     * @throws GeoLocationDataException
+     */
+    @Messages({
+        "# {0} - area name",
+        "GEOArea_point_label_header=Area outline point for area: {0}"
+    })
+    private void buildPath(GeoAreaPoints points, BlackboardArtifact artifact) throws GeoLocationDataException {
+        for (GeoAreaPoints.AreaPoint point : points) {
+            addToPath(new AreaWaypoint(artifact, Bundle.GEOArea_point_label_header(getLabel()), point));
+        }
+    }
+
+    /**
+     * Returns the list of GeoAreaPoints from the attributeMap. Creates the
+     * GeoAreaPoint list from the TSK_GEO_AREAPOINTS attribute.
+     *
+     * @param attributeMap Map of artifact attributes.
+     *
+     * @return GeoTrackPoint list empty list if the attribute was not found.
+     *
+     * @throws GeoLocationDataException
+     */
+    private GeoAreaPoints getPointsList(Map<BlackboardAttribute.ATTRIBUTE_TYPE, BlackboardAttribute> attributeMap) throws GeoLocationDataException {
+        BlackboardAttribute attribute = attributeMap.get(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_AREAPOINTS);
+        if (attribute == null) {
+            LOGGER.log(Level.SEVERE, "No TSK_GEO_AREAPOINTS attribute was present on the artifact.");
```

Comment:
Do we need to log the error and throw an exception?  Will the calling code write an error too?

Reply:
So that's copied from the Track class, and the error actually isn't logged in the one place I see it being caught in GeoPath.
} catch (GeoLocationDataException e) {
   allParsedSuccessfully = false;
}
But I think it is better behavior to throw the exception and then log it up there where we have access to the artifactID, so I'll update my code and the Track code.